src/functions/pollJournals/framework/repo/mysql/row-mappers/test-slot-row-mapper.ts

```diff
@@ -0,0 +1,141 @@
+import { ExaminerTestSlot } from '../../../../domain/examiner-test-slot';
+import { VehicleGearbox, PreviousCancellation } from '../../../../../../common/domain/Schema';
+
+interface TestSlotRow {
+  slot_id: number;
+  start_time: string;
+  minutes: number;
+  vehicle_slot_type: string;
+  tc_id: number;
+  tc_cost_centre_code: string;
+  tc_name: string;
+  individual_id: number;
+  programme_date: string;
+  booking_id: number;
+  app_id: number;
+  booking_seq: number;
+  check_digit: number;
+  welsh_test_ind: number;
+  ext_req_ind: number;
+  progressive_access: number;
+  meeting_place: string;
+  special_needs: string;
+  ent_check_ind: number;
+  cab_seat_count: number;
+  passenger_seat_count: number;
+  height_metres: number;
+  length_metres: number;
+  width_metres: number;
+  vehicle_category: string;
+  gearbox_type: string;
+  candidate_id: number;
+  candidate_title: string;
+  candidate_first_name: string;
+  candidate_second_name: string;
+  candidate_third_name: string;
+  candidate_surname: string;
+  candidate_driver_number: string;
+  cand_primary_tel: string;
+  cand_secondary_tel: string;
+  cand_mobile_tel: string;
+  cand_email: string;
+  candidate_addr_line1: string;
+  candidate_addr_line2: string;
+  candidate_addr_line3: string;
+  candidate_addr_line4: string;
+  candidate_addr_line5: string;
+  candidate_post_code: string;
+  candidate_prn: number;
+  prev_attempts: number;
+  business_id: number;
+  business_name: string;
+  business_addr_line1: string;
+  business_addr_line2: string;
+  business_addr_line3: string;
+  business_addr_line4: string;
+  business_addr_line5: string;
+  business_post_code: string;
+  business_telephone: string;
+  cancel_initiator: string;
+}
+
+export const mapRow = (row: TestSlotRow): ExaminerTestSlot => {
+  return {
+    examinerId: row.individual_id,
+    testSlot: {
+      slotDetail: {
+        slotId: row.slot_id,
+        start: row.start_time,
+        duration: row.minutes,
+      },
+      vehicleSlotType: row.vehicle_slot_type,
+      testCentre: {
+        centreId: row.tc_id,
+        centreName: row.tc_name,
+        costCode: row.tc_cost_centre_code,
+      },
+      booking: {
+        candidate: {
+          candidateId: row.candidate_id,
+          candidateName: {
+            title: row.candidate_title,
+            firstName: row.candidate_first_name,
+            secondName: row.candidate_second_name,
+            thirdName: row.candidate_third_name,
+            lastName: row.candidate_surname,
+          },
+          driverNumber: row.candidate_driver_number,
+          candidateAddress: {
+            addressLine1: row.candidate_addr_line1,
+            addressLine2: row.candidate_addr_line2,
+            addressLine3: row.candidate_addr_line3,
+            addressLine4: row.candidate_addr_line4,
+            addressLine5: row.candidate_addr_line5,
```

Comment:
Would be handy to know what are these address lines actually hold and name them accordingly.
i.e city, county, street etc.

Reply:
Not exactly sure what change you're suggesting here. Changing the schema to have fields like `town` instead of `addressLine1`?